garrysmod/lua/menu/openurl.lua

```diff
@@ -38,6 +38,100 @@ function GMOD_OpenURLNoOverlay( url )
 
 end
 
+----------------------------------------------
+-- patch by @catSIXe - can be used in orig. gmod code if wanted
+----------------------------------------------
+
+-- better code, but we should try to save this somewhere where only the menu context can write to(read should be no problem lol)
+local _whitelistFilePath = "_connectwhitelist.json"
+local _whitelistDir = "DATA"
+local function parseAddress(serverip)
```

Comment:
your code is cringe, make it more like this
```
local function ParseAddress(ip)
    local host, port = unpack(ip:Split(":"))
    port = tonumber(port) or 27015

    if port ~= math.Clamp(math.floor(port), 1, 65535)) then return end

    local ipbytes = host:Split(".")
    if #ipbytes ~= 4 then return end
    for _,v in ipairs(ipbytes) do
        v = tonumber(v)
        if v ~= math.Clamp(v or 0, 0, 255) then return end
    end

    return host, port
end
```

Reply:
i hope that was satire